### PR TITLE
✔️ Integration tests

### PR DESCRIPTION
**Pull request by JUtupe**

### CODE REVIEW

File: src/test/kotlin/pl/newbies/auth/GithubAuthTest.kt [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    @Nested
+    inner class Login {
+        @Test
+        fun `should redirect when login called`() = withAres {
+            // when
+            val response = httpClient.get("/oauth/login/github")
+
+            // then
+            assertEquals(HttpStatusCode.OK, response.status)
```
Comment (DiD3n) [inline src/test/kotlin/pl/newbies/auth/GithubAuthTest.kt:27]: *where* redirect check 
it looks like you are only checking if the request is executing, not how

File: src/test/kotlin/pl/newbies/auth/GithubAuthTest.kt [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        fun `should create account when called with new user`() = withAres {
+            // given
+            clearTable("Users")
+            val user = TestData.testUser1
+
+            // when
+            val response = loginAs(user)
+
+            // then
+            assertEquals(user.name, response.username)
```
Comment (DiD3n) [inline src/test/kotlin/pl/newbies/auth/GithubAuthTest.kt:44]: ok, user went through 
what's about looking in db if he really was created

File: src/test/kotlin/pl/newbies/auth/GithubAuthTest.kt [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        }
+
+        @Test
+        fun `should return existing account when called with existing github user`() = withAres {
+            // given
+            clearTable("Users")
```
Comment (DiD3n) [inline src/test/kotlin/pl/newbies/auth/GithubAuthTest.kt:51]: i can't determine if user already exists bc this test looks exactly the same as one above

**Review by DiD3n — APPROVED** [review]

File: src/test/kotlin/pl/newbies/auth/GithubAuthTest.kt (hunk 3 above)
Comment (JUtupe) [reply src/test/kotlin/pl/newbies/auth/GithubAuthTest.kt:51]: Theoretically user should be created when first `loginAs(...)` is called, this test call loginAs twice to check if the second call actually returns user, maybe I should do some "createDate" comparison to determine if it is actually the previous user

File: src/test/kotlin/pl/newbies/auth/GithubAuthTest.kt (hunk 2 above)
Comment (JUtupe) [reply src/test/kotlin/pl/newbies/auth/GithubAuthTest.kt:44]: I'll do it by using GET method

File: src/test/kotlin/pl/newbies/auth/GithubAuthTest.kt (hunk 3 above)
Comment (DiD3n) [reply src/test/kotlin/pl/newbies/auth/GithubAuthTest.kt:51]: my bad
didn't see second one

File: src/test/kotlin/pl/newbies/auth/GithubAuthTest.kt (hunk 1 above)
Comment (JUtupe) [reply src/test/kotlin/pl/newbies/auth/GithubAuthTest.kt:27]: Github has a mocked module, which returns "ok" as a content on expected (see line below and githubModule() in TestModules.kt)

File: src/test/kotlin/pl/newbies/auth/GithubAuthTest.kt (hunk 3 above)
Comment (JUtupe) [reply src/test/kotlin/pl/newbies/auth/GithubAuthTest.kt:51]: I've added assertions to check if createDate is the same after second login, that should clarify that user data is actually persisted 😄

File: src/test/kotlin/pl/newbies/auth/GithubAuthTest.kt (hunk 2 above)
Comment (JUtupe) [reply src/test/kotlin/pl/newbies/auth/GithubAuthTest.kt:44]: done

**Pull request merged**